OpenRA.Mods.Common/AI/HackyAI.cs

```diff
@@ -82,6 +95,11 @@ public sealed class HackyAIInfo : IBotInfo, ITraitInfo
 		[Desc("Radius in cells around the center of the base to expand.")]
 		public readonly int MaxBaseRadius = 20;
 
+		[Desc("Radius in cells around each building with ProvideBuildableArea",
+		"to check for a 3x3 area of water where naval structures can be built.",
```

Comment:
@penev92: It explicitly does not check the footprint indeed, however it counts how many water tiles surrounded by fully buildable water tiles exist, which ultimately does end up with a 3x3 buildable water area.

@reaperrr: Please add TODOs into `EnoughWaterToBuildNaval()` and `CloseEnoughToWater()` then at least.